src/app/shared/markdown-wrapper/markdown-wrapper.component.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterViewInit, Component, Input, OnInit, ViewChild } from '@angular/core';
+import DOMPurify from 'dompurify';
+import { MarkdownComponent } from 'ngx-markdown';
+
+@Component({
+  selector: 'markdown-wrapper',
+  templateUrl: './markdown-wrapper.component.html',
+  styleUrls: ['./markdown-wrapper.component.scss'],
+})
+
+// Wrapper for user-input markdown to provide custom sanitization not accessible through the built-in sanitizer, DOMSanitize.
+// DOMSanitize's safe-lists can be seen https://github.com/angular/angular/blob/master/packages/core/src/sanitization/html_sanitizer.ts
+export class MarkdownWrapperComponent implements OnInit, AfterViewInit {
+  @ViewChild(MarkdownComponent) mkdRef;
+  @Input() data: string;
+  forbid_tags: string[];
+  forbid_attr: string[];
+
+  ngOnInit() {
+    // tags and attributes that will be sanitized by DOMPurify.
+    this.forbid_tags = [];
+    this.forbid_attr = ['class'];
+
+    // Sets config for DOMPurify until otherwise specified.
+    DOMPurify.setConfig({
+      FORBID_TAGS: this.forbid_tags,
+      FORBID_ATTR: this.forbid_attr,
+    });
+  }
+
+  ngAfterViewInit() {
```

Comment:
Does this work if you edit the markdown? Not an Angular life-cycle expert, but I am concerned that that this may only execute once. If you're the owner of the org and edit the content, what happens?

I could  be wrong on this once; it's possible the control gets recreated. But best to make sure.

While I don't think it would be a security risk -- the only person seeing the changes would the self-same author, it does have the potential of displaying something differently than they will ultimately be (the classes would be briefly active).

Reply:
Updated in newer commit. After looking more into this, I noticed the MarkdownModule provided a few more features than I initially thought. Using the MarkdownService, I am able to compile the input variable from markdown to html, and then pass that to DOMPurify. See [here](https://github.com/dockstore/dockstore-ui2/blob/727f3ddd82101763d57a954cb1d9a85122a76a8b/src/app/shared/markdown-wrapper/markdown-wrapper.component.ts#L39). This will now be updated whenever a change is detected on input.